### PR TITLE
support validator array

### PR DESCRIPTION
**Pull request by nicolazj**
So we can do something like this:

```
  schema=[
      (`email, [Required, Email]),
      (`password, [Required]),
    ]
```

### CODE REVIEW

**Comment by grsabreu** [comment]
Awesome, thanks for implementing this! We were thinking about it for a while indeed
I'll fix the CI script and review soon your code

File: re/ReForm_Validation.re [hunk 1]
```diff
@@ -3,25 +3,59 @@
...
-  switch validator {
-  | Required => String.length(value) < 1 ? Some(i18n.required) : None
+    (validator, ~values, ~value, ~i18n: I18n.dictionary)
+    : error =>
+  switch (validator) {
+  | Required => String.length(value) > 0 ? None : Some(i18n.required)
   | Custom(fn) => fn(values)
   | Email =>
-    Js.Re.test(value, [%bs.re {|/\S+@\S+\.\S+/|}]) ? None : Some(i18n.email)
+    String.length(value) == 0
```
Comment (grsabreu) [inline re/ReForm_Validation.re]: I think `String.length(value) == 0` should be kept out, if the string is empty it is indeed an error

File: re/ReForm_Validation.re [hunk 2]
```diff
@@ -3,25 +3,59 @@
...
+type errors = option(list(string));
+
 type validation('values) =
   | Required
   | Email
-  | Custom('values => option(string));
+  | Custom('values => error);
+
+/* turn list(error) => errors */
+let squeeze = (errorList: list(error)) => {
```
Comment (grsabreu) [inline re/ReForm_Validation.re]: hmm I think we just should return the list of errors to the user

Comment (nicolazj) [reply re/ReForm_Validation.re]: Do you mean list(string) ?

File: re/ReForm_Validation.re (hunk 1 above)
Comment (nicolazj) [reply re/ReForm_Validation.re]: how to deal with this situation? the field is optional, but if it's not empty string, it should be a valid email ?

Comment (grsabreu) [reply re/ReForm_Validation.re]: Ah, I got your point hmm I think we can let like that indeed for now

**Pull request closed without merging**